naloge/bfs.tex

```diff
@@ -10,6 +11,112 @@
 \caption{Graf za nalogi~\nal in~\nal[dfs].}
 \end{slika}
 \end{vprasanje}
+
 \begin{odgovor}
+ALGORITEM:\\
+\underline{VHOD}:  graf G, vozlišče $s \in V$\\
+\underline{IZHOD}: razdalje $d_G(s, v)$ za vsak $v \in V$\\
+def bfs(G, s):\\
+	\phantom{x}\hspace{3ex}za vsak $v \in V(G)$:\\
+		\phantom{x}\hspace{6ex}razdalja[v] = $\infty$\\
+	\phantom{x}\hspace{3ex}razdalja[s] = 0\\
+	\phantom{x}\hspace{3ex}Q = [s]\\
+	\phantom{x}\hspace{3ex}dokler Q ni prazna:\\
+		\phantom{x}\hspace{6ex}u = odstrani-prvega(Q)\\
+		\phantom{x}\hspace{6ex}za vsak $(u, v) \in E(G)$:\\
+			\phantom{x}\hspace{9ex}če razdalja[v] = $\infty$:\\
+				\phantom{x}\hspace{12ex}razdalja[v] = razdalja[u] + 1\\
+				\phantom{x}\hspace{12ex}vstavi(Q, v)\\
+\begin{flushleft}
+Če sledimo zgornjemu algoritmu in predpostavki, da upoštevamo vrstni red, začnemo v vozlišču A, je potek sledeč: 
+\end{flushleft}
+\underline{Predpriprava:}\\
+razdalja = $[\infty, \infty, \infty, \infty, \infty, \infty]$\\
+razdalja[A] = 0\\
+Q = [A]\\
+\\
+\underline{1. korak:}\\
+Odstranimo A iz vrste in pregledamo vse njegove sosede.\\
+razdalja[B] = 0 + 1 = 1\\
+Q = [B]\\
+razdalja[E] = 1\\
+Q = [B, E]\\
+\\
+\underline{2. korak:}\\
+Odstranimo B iz vrste in pogledamo vse njegove sosede, ki so še vedno na razdalji $\infty$.\\
+razdalja[C] = 1 + 1 = 2\\
+Q = [E, C]\\
+\\
+\underline{3.korak:}\\
+Odstranimo E iz vrste.\\
+razdalja[F] = 1 + 1 = 2\\
+Q = [C, F]\\
+\\
+\underline{4. korak:}\\
+Odstranimo C iz vrste. Ker nima sosedov, ki bi imeli razdaljo še vedno nastavljeno na $\infty$, gremo naprej.\\
+Q = [F]\\
+\\
+\underline{6. korak:}\\
+Odstranimo F iz vrste.\\
+razdalja[I] = 3\\
+Q = [I]\\
+\\
+\underline{6. korak:}\\
+Odstranimo I iz vrste. Ker nima sosedov na razdalji $\infty$, gremo naprej.\\
+\\
+\underline{7. korak:}\\
+Ker je na tej točki vrsta prazna, se algoritem ustavi.\\
+\\
+\begin{flushleft}
+Podobno naredimo za drugi del grafa. Ker nista povezana, moramo algoritem začeti od začetka. 
+\end{flushleft}
+\underline{Predpriprava:}\\
+razdalja = $[\infty, \infty, \infty]$\\
+razdalja[D] = 0\\
+Q = [D]\\
+\\
+\underline{1. korak:}\\
+Iz vrste odstranimo D.\\
+razdalja[G] = 1\\
+Q = [G]\\
+razdalja[H] = 1\\
+Q = [G, H]\\
+\\
+\underline{2. korak:}\\
+Odstranimo G iz vrste. Ker nima sosedov na razdalji $\infty$, gremo naprej.\\
+\\
+\underline{3. korak:}\\
+Odstranimo H iz vrste. Ker nima sosedov na razdalji $\infty$, gremo naprej.\\
+\\
+\underline{4. korak:}\\
+Ker je na tej točki vrsta prazna, se algoritem ustavi.
+\newpage
+\begin{flushleft}
+Drevo iskanja v širino: 
+\end{flushleft}
+\begin{tikzpicture}[style=thick,scale=0.7]
```

Comment:
Prosil bi, da sliko (torej vse od `\begin{tikzpicture}` do `\end{tikzpicture}`) premakneš v svojo datoteko v mapi `slike/` - glede na to, da ta naloga že ima svojo sliko, svetujem, da datoteko poimenuješ `bfs-resitev.tikz`. Sliko lahko potem vključiš z
```LaTeX
\begin{slika}
\pgfslika[bfs-resitev]
\podnaslov{Gozd iskanja v širino}
\end{slika}
```
V besedilu rešitve se potem na sliko sklicuješ s `\fig[bfs-resitev]`.

Na tak način je mogoče sliko tudi posebej prevesti, da se potem sama zbirka hitreje prevaja.